1_fetch.R

```diff
@@ -41,11 +41,15 @@ p1_targets_list <- list(
   tar_target(
     p1_GFv1_catchments_sf,
     {
-      sf::st_read(dsn = p1_GFv1_catchments_shp,layer="nhru", quiet=TRUE) %>%
-        filter(hru_segment %in% p1_GFv1_reaches_sf$subsegseg) %>%
-        suppressWarnings() %>%
-        # fix geometry issues by defining a zero-width buffer around the polylines
-        sf::st_buffer(.,0)
+      hru_layer_name <- grep("hru", sf::st_layers(p1_GFv1_catchments_shp)$name, value=TRUE)
```

Comment:
Are you expecting this layer name to change? Or is it more than one layer name? Just wondering why you are assigning this dynamically instead of hard-coding it in? 

If it's just one value, I think hard-coding it would be better for clarity and so that you aren't accidently matching a layer that you aren't expecting.

If there needs to be this dynamic assignment, I think a one-line comment to explain a couple of matches you are expecting would be could help when looking at this down the road.

Reply:
Good point, I agree that this isn't very easy to read. I assign the HRU layer name dynamically here because I wanted to keep the pipeline flexible to downloading either the national-scale data or the regional subsets, and those versions use different names for the shp that corresponds to the HRU's. Because of that, I propose keeping the dynamic assignment as is. However, in https://github.com/USGS-R/drb-network-prep/pull/7/commits/910cc43d4be1e70ce66e6ab894de9637d4d78289 I added a comment as well as a line of code that stops the target from building if the layer name does not match one of two expected values. Hopefully that will help avoid a situation where we're matching a layer that we're not expecting.